frontend/src/components/Input/Input.tsx

```diff
@@ -0,0 +1,52 @@
+import React, {useState} from 'react';
+import {
+  StyledInput,
+  StyledInputWrapper,
+  StyledLabel,
+  StyledErrorMessage,
+} from './Input.styled';
+
+interface Props {
+  disabled: boolean;
+  label: string;
+  withError: boolean;
+  placeholder: string;
+  errorMessage: string;
+}
+
+const Input: React.FC<Props> = ({
+  disabled,
+  label,
+  withError,
+  placeholder,
+  errorMessage,
+}) => {
+
+const [inputValue, setInputValue] = useState('')
```

Comment:
Added type into useState.
But still have some errors with props in Input Component and also in Input.styled.tsc. I do not uderstand what is the problem :/ 
<img width="987" alt="Zrzut ekranu 2021-03-23 o 21 31 31" src="https://user-images.githubusercontent.com/59742755/112214676-c3b18d00-8c1f-11eb-9a11-62da65025406.png">
<img width="1041" alt="Zrzut ekranu 2021-03-23 o 21 30 18" src="https://user-images.githubusercontent.com/59742755/112214685-c6ac7d80-8c1f-11eb-899a-1fc0412f9bc8.png">

Reply:
> please add type to useState, like eg const [state, setState] = useState<string>('');

TS is not that dumb, it can infer type of state by value used as initial value 😄 
Could you please remove that type? It's not needed

Reply:
This issue exists because you didn't tell TypeScript that e.g. `StyledInput` could take additional props, like `withError`. `disabled` attribute is not supported by `label` element, here's the reference https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/disabled

> The disabled attribute is supported by <button>, <command>, <fieldset>, <keygen>, <optgroup>, <option>, <select>, <textarea> and <input>.

To resolve this issue you'd have to specify type of additional props. To give you an example:

```ts
interface StyledLabelProps {
  withError: boolean;
  disabled: boolean;
}

export const StyledLabel = styled.label<StyledLabelProps>`
  font-size: 10px;
  margin-bottom: 2%;
  ${(props) => {
    // Now we can use `withError` and `disabled` since we've
    // explicitly said that this component have additional properties
    // both of type `boolean`: `withError` and `disabled`
    if (props.withError) {
      return `color: #9C1B07`;
    }

    if (props.disabled) {
      return `color: #D1D1D1`;
    }

    return `color: black`;
  }}
`;
```

If you don't understand what's that strange symbols after `styled.label`, don't worry, you'll get it 😄
To clear some confusion: function `styled.label` can take some generic types. You pass those types to `styled.label` by using `<  >` syntax - it's similar to JSX or HTML. Between those sharp brackets you can pass any type. In TypeScript or just in programming world 🌎 this concept is called `generics`. You can read more about it here https://www.typescriptlang.org/docs/handbook/2/generics.html 

Reference for specifying additional props on styled components: https://styled-components.com/docs/api#using-custom-props